ReplicatorHelper.swift

```diff
@@ -114,103 +305,499 @@ public struct ReplicatorHelper {
-            case "PUSH":
-                replConfig.replicatorType = .push
-            default:
-                throw ReplicatorError.fatalError(message: "Invalid replicatorType")
-        }
         return replConfig
     }
     
-    public static func replicatorCollectionConfigFromJson(_ data:  [CollectionConfigItem], replicationConfig: inout ReplicatorConfiguration) throws {
+    /// **[OLD API SUPPORT]** Creates ReplicatorConfiguration from OLD API format
+    ///
+    /// **What it does:**
+    /// - Processes OLD API format where multiple collections can share one config
+    /// - Creates ReplicatorConfiguration using endpoint-only constructor
+    /// - Adds collections with their shared config via mutation
+    /// - Maintains backward compatibility with existing implementations
+    ///
+    /// **Parameters:**
+    /// - `data`: Dictionary containing all replicator settings
+    /// - `collectionConfiguration`: Array of CollectionConfigItem (OLD API format)
+    ///
+    /// **OLD API Format:**
+    /// ```swift
+    /// CollectionConfigItem(
+    ///   collections: [CollectionDtoWrapper(collection: CollectionDto(...))],
+    ///   config: ConfigDto(channels: [...], ...)
+    /// )
+    /// ```
+    ///
+    /// **Throws:**
+    /// - `ReplicatorError.fatalError`: If required fields are missing or invalid
+    /// - `CollectionError`: If collection lookup fails
+    private static func replicatorConfigFromJsonOldApi(
+        _ data: [String: Any],
+        collectionConfiguration: [CollectionConfigItem]
+    ) throws -> ReplicatorConfiguration {
+        print("\n[OLD API] ═══════════════════════════════════════════════════════")
+        print("[OLD API] Starting OLD API ReplicatorConfiguration creation")
+        print("[OLD API] ═══════════════════════════════════════════════════════")
         
-        //work on the collections sent in as part of the configuration with an array of collectionName, scopeName, and databaseName
-        for item in data {
-            
+        // Parse and validate required fields
+        print("[OLD API Step 1] Parsing required fields from data dictionary...")
+        guard let target = data["target"] as? [String: Any],
+              let url = target["url"] as? String,
+              let replicatorType = data["replicatorType"] as? String,
+              let continuous = data["continuous"] as? Bool,
+              let acceptParentDomainCookies = data["acceptParentDomainCookies"] as? Bool,
+              let acceptSelfSignedCerts = data["acceptSelfSignedCerts"] as? Bool,
+              let allowReplicationInBackground = data["allowReplicationInBackground"] as? Bool,
+              let autoPurgeEnabled = data["autoPurgeEnabled"] as? Bool,
+              let heartbeat = data["heartbeat"] as? NSNumber,
+              let maxAttempts = data["maxAttempts"] as? NSNumber,
+              let maxAttemptWaitTime = data["maxAttemptWaitTime"] as? NSNumber,
```

Comment:
I am not sure about corresponding types on JS for TimeInterval, but if you are looking for ObjC types here, these are the native CBL's:
```
NSTimeInterval heartbeat;
NSUInteger maxAttempts;
NSTimeInterval maxAttemptWaitTime;
```

While for Swift - already reflected down the line when calling setting up the CBL args:
```
heartbeat: TimeInterval
maxAttempts: UInt
maxAttemptWaitTime: TimeInterval
```

I am wondering if we can avoid converting number values as much as possible, what is the expected value from the user end on RN end? Can we expect something else than a generic `NSNumber` here?

CC @pasin for thoughts on this and the conversion below